### ISSUE TITLE
[FEATURE]: make it possible to detect if devMode is active for the current module in order to use this information in templating

### ISSUE THREAD

**Issue by sundbp** [garden-io/garden#3552]
## Feature Request

I propose to add a way for a module context to include information about if devMode is active or not (runtime resolved), and enable module templating to rely on this information.

### Background / Motivation

Common problem: the production image for a service does not contain the required dev tooling one may need for devMode (e.g. debuggers and other such dev tooling). In order to not touch the production image, while still providing an excellent devMode experience, it'd be great to use an alternative image in devMode that contains all the goodies.

If we could use e.g. conditionals on the devMode being active or not within a module template, this would make this simple to  define.

### What should the user be able to do?

The user would be able to provide a clean separation between production/deploy style images and devMode images, and experience the best of both worlds. In particular, the freedom to create a really awesome devMode experience without impacting any production like flows is increased.

### Why do they want to do this? What problem does it solve?

The main pain point is to allow for a richer devMode experience without disrupting production workflows.

### Suggested Implementation(s)

If all module contexts contained information about if devMode is active or not we could rely on this in template conditionals in the module definition itself.

### How important is this feature for you/your team?

🌵 Not having this feature makes using Garden painful

**Comment by Orzelius** [comment]
Hello, this is a good idea. As a workaround you could use something like this
```yaml
variables:
  devModeSettings:
    foo: bar
  normalModeSettings: {}
values:
  $merge: "${command.params contains 'dev-mode' ? var.devModeSettings : var.normalModeSettings}"
```

**Comment by sundbp** [comment]
Or to be specific about dev-mode being applied to the current service:
`${command.params contains 'dev-mode' && command.params.dev-mode contains 'my-service'}`

The feature request still makes sense to me though, as this workaround means string matching, so if we change module.name we also need to remember to go and change the string matching. A devMode boolean in the module's context would be cleaner (and I guess the same could be said for similar module runtime state like local mode, hot-reload, etc).

**Comment by sundbp** [comment]
btw, I think this is a good illustration that the template features of garden has a lot of power. Instead of needing special support/feature to separate devMode and non-devMode images, we can rely on template "programming". It's still easy to follow the logic when reading such a template so I think the flexibility doesn't come at too high cost of complexity.

Nicely done!

**Comment by vvagaytsev** [comment]
@sundbp this has been addressed in #3913. So, it will be released in the `0.13.0`.
